### DISCUSSION TITLE
many connection_init

### DISCUSSION THREAD

**Discussion by baonv-acbs-com-vn** [enisdenjo/graphql-ws#498] [Q&A] [answered]
I noticed that the client sends another connection_init after the connection has been established
![Screen Shot 2023-08-19 at 12 38 08](https://github.com/enisdenjo/graphql-ws/assets/110224931/f7105e94-383d-4c90-8df2-739dcf29cb74)
![Screen Shot 2023-08-19 at 12 38 20](https://github.com/enisdenjo/graphql-ws/assets/110224931/22cfc411-ed1b-403a-b7c4-35ae43fb3973)
is there a way to switch back to one?

**Comment by enisdenjo** [comment, marked as answer]
> If the server receives more than one `ConnectionInit` message at any given time, the server will close the socket with the event `4429: Too many initialisation requests`.

[_GraphQL over WebSocket spec_](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md#connectioninit)

If the client sent multiple, the server would've terminated the connection immediately. So this is not happening. The screenshots you provided showcase two separate WS connections (see the selection on the left), each connection has only one `connection_init` message.

You're most likely instantiating multiple `graphql-ws` clients.
